[Merge Pending] Allow passing query between egress filter and filter gateway.

kevindiu opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description:

<!-- Describe your changes in detail -->
<!-- It would be better to describe the details especially What changed and Why you changed -->

This PR updated the following changes:

- Allow passing `query` to egress filter through `FilterDistance` and `FilterVector` API 
  - See `apis/proto/v1/filter/egress/egress_filter.proto`
- Update Search APIs to allow filter searching result using query
  - For the logic, see `pkg/gateway/filter/handler/grpc/handler.go`
- Updated existing Insert, Update, Upsert, GetObject API to allow filter object before corresponding operation
  - Updated to allow passing different query through multiple filtering target
- Add example code, including example egress filter implementation, and example client code to connect to filter gateway using query filter feature.
  - See `example/client/gateway/filter/README.md` 
- Update related document about above changes.

### Related Issue:

<!-- This project mainly accepts pull requests related to open issues -->
<!-- NOTE: If suggesting a new feature or change, please discuss it in an issue first -->
<!-- NOTE: If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!-- Please link to the issue here: -->

### Versions:

<!--- Please change the versions below along with your environment -->

- Go Version: 1.20.3
- Docker Version: 23.0.5
- Kubernetes Version: 1.27.1
- NGT Version: 2.0.9

### Checklist:

<!-- For completed items, change [ ] to [x]. -->
<!-- NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open. -->

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### Special notes for your reviewer:

<!-- Please tell us anything you would like to share to reviewers related this PR -->

You can check the behavior after this modification by reading the `example/client/gateway/filter/README.md `.

You may also need to edit the below file to enable filter gateway and setting the image tag to validate the implementation.
```
vim example/helm/values.yaml

gateway:
  filter:
    enabled: true
    image:
      tag: pr-2097
```

Please do not merge this PR, unless the target merge branch is updated.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for new filter payload types: DistanceRequest, DistanceResponse, Query, VectorRequest, and VectorResponse.
  - Introduced query fields in filter configurations for enhanced filtering capabilities.
  - Implemented new methods for filtering distances and vectors.
  - Enhanced search functionality to support multiple ingress and egress filters.
  - Added aggregation algorithm options in search configurations.

- **Documentation**
  - Updated API documentation to reflect new filter payload types and query fields.
  - Enhanced user guides with new filter configuration options and examples, including details on ingress and egress filters.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mtanig committed:
Allow passing query between egress filter and filter gateway. (#1896)

* Update payload.proto.

* Change Filter.Config.

* Change prototypes of egress filter.

* Update proto for Insert, Upsert, Update and Object.

* Add imprements in handler.

* Fix FilterVector and FilterDistance in Client.

* Use net.JointHostPost instead of fmt.Sprintf .

* Regenerate pb files and format all.

* Add an example for current filtering.

* Add example ingress-filter and egress-filter for checking the feature of filtering.

* Fix manifests and format codes.

* Delete debug code.

* Delete unused import.

* Fix filtering tests to work correctly with current version.

* Add a README for example.

* Fix example to work at e2e level.

* Modify README.

* Update README to work in k3d.

* Fix k8s.mk .

* Fix README.

* Add reference to examples.

* Fix to pass the format command.

* Change Object.Vector in VectorRequest/Response from array to singular.

* Modify docs.

* Fix makefiles and dockerfiles.

* Fix k8s.mk .

* Fix error messages in filter gateway.

* Run make proto/all && make format, but in a valid developer's environment the same command is run after that.

* make proto/all && make format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

* regenerate proto

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

* draft: support default filter config

Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>

* fix

Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>

* fix

Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>

---------

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>
Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>
Co-authored-by: Yusuke Kato <kpango@vdaas.org>
Co-authored-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>
Co-authored-by: kevindiu <kevin_diu@yahoo.com.hk>
Co-authored-by: Kevin Diu <kevindiujp@gmail.com>

deepsource-autofix[bot] committed:
style: Format code with prettier and gofumpt

kevindiu committed:
update config type

kevindiu commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,67 @@
+2. Build and publish example ingress filter and egress filter docker image
+
+```bash
+# login to docker if need, and setup your DockerHub ID
````
```suggestion
# login to docker if needed, and setup your DockerHub ID
```

kevindiu commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,67 @@
+2. Build and publish example ingress filter and egress filter docker image
+
+```bash
+# login to docker if need, and setup your DockerHub ID
````
```suggestion
# login to docker if needed, and setup your DockerHub ID
```

kevindiu commented:
I force pushed twice to rebase the branch to main branch.
The first force push `3415596 to 92a7f0e` is only squash commits in this branch without any changes, to easier rebase the branch.
The second force push `92a7f0e to 4e437a5` is the actual rebase command.

kevindiu committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

kevindiu committed:
fix example client

Signed-off-by: kevindiu <kevin_diu@yahoo.com.hk>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
refactor: example code

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: execute make format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: refactor example code

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

vankichi commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,73 @@
+After launching the k8s cluster, do the following steps at the root directory.
+
+1. Deploy Vald cluster with filter-gateway
+```bash
+vim example/helm/values.yaml
+---
+...
+gateway:
+...
+    filter:
+        enabled: true
+...
+agent:
+    ngt:
+        dimension: 784
+        distance_type: l2
+...
+
+# deploy vald cluster
+helm repo add vald https://vald.vdaas.org/charts
+helm install vald vald/vald --values example/helm/values.yaml
+```
````
```suggestion

      ```bash
      vim example/helm/values.yaml
      ---
      ...
      gateway:
      ...
      filter:
          enabled: true
      ...
      agent:
          ngt:
              dimension: 784
              distance_type: l2
      ...
      
      # deploy vald cluster
      helm repo add vald https://vald.vdaas.org/charts
      helm install vald vald/vald --values example/helm/values.yaml
      ```
```

hlts2 replied:
Thank you for the suggestion 🙏 
I will apply these changes!

vankichi commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,73 @@
+
+2. Build and publish example ingress filter and egress filter docker image
+
+```bash
+# login to docker if needed, and setup your DockerHub ID
+docker login
+export DOCKERHUB_ID=<DOCKERHUB_ID>
+
+# build and publish ingress filter image
+docker build \
+    -f example/manifest/filter/ingress/Dockerfile \
+    -t $DOCKERHUB_ID/vald-ingress-filter:latest . \
+    --build-arg GO_VERSION=$(make version/go)
+
+docker push ${DOCKERHUB_ID}/vald-ingress-filter:latest
+
+# build and publish egress filter image
+docker build \
+    -f example/manifest/filter/egress/Dockerfile \
+    -t $DOCKERHUB_ID/vald-egress-filter:latest . \
+    --build-arg GO_VERSION=$(make version/go)
+
+docker push ${DOCKERHUB_ID}/vald-egress-filter:latest
+```
````
```suggestion
    ```bash
    # login to docker if needed, and setup your DockerHub ID
    docker login
    export DOCKERHUB_ID=<DOCKERHUB_ID>
    
    # build and publish ingress filter image
    docker build \
        -f example/manifest/filter/ingress/Dockerfile \
        -t $DOCKERHUB_ID/vald-ingress-filter:latest . \
        --build-arg GO_VERSION=$(make version/go)
    
    docker push ${DOCKERHUB_ID}/vald-ingress-filter:latest
    
    # build and publish egress filter image
    docker build \
        -f example/manifest/filter/egress/Dockerfile \
        -t $DOCKERHUB_ID/vald-egress-filter:latest . \
        --build-arg GO_VERSION=$(make version/go)
    
    docker push ${DOCKERHUB_ID}/vald-egress-filter:latest
    ```
```

vankichi commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,73 @@
+```
+
+3. Deploy ingress filter server and egress filter server
+```bash
+# deploy ingress filter
+sed -e "s/DOCKERHUB_ID/${DOCKERHUB_ID}/g" example/manifest/filter/egress/deployment.yaml | kubectl apply -f - \
+&& kubectl apply -f example/manifest/filter/egress/service.yaml
+
+# deploy egress filter
+sed -e "s/DOCKERHUB_ID/${DOCKERHUB_ID}/g" example/manifest/filter/ingress/deployment.yaml | kubectl apply -f - \
+&& kubectl apply -f example/manifest/filter/ingress/service.yaml
+```
````
```suggestion

    ```bash
    # deploy ingress filter
    sed -e "s/DOCKERHUB_ID/${DOCKERHUB_ID}/g" example/manifest/filter/egress/deployment.yaml | kubectl apply -f - \
    && kubectl apply -f example/manifest/filter/egress/service.yaml
    
    # deploy egress filter
    sed -e "s/DOCKERHUB_ID/${DOCKERHUB_ID}/g" example/manifest/filter/ingress/deployment.yaml | kubectl apply -f - \
    && kubectl apply -f example/manifest/filter/ingress/service.yaml
    ```
```

vankichi commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,73 @@
+```
+
+4. Run test
+```bash
+# if you don't use the Kubernetes ingress, set the port forward
+kubectl port-forward deployment/vald-filter-gateway 8081:8081
+
+# Please change the argument according to your environment.
+go run ./example/client/gateway/filter/main.go -addr "localhost:8081" -ingresshost "vald-ingress-filter.default.svc.cluster.local" -ingressport 8082 -egresshost "vald-egress-filter.default.svc.cluster.local" -egressport 8083
+```
````
```suggestion

    ```bash
    # if you don't use the Kubernetes ingress, set the port forward
    kubectl port-forward deployment/vald-filter-gateway 8081:8081
    
    # Please change the argument according to your environment.
    go run ./example/client/gateway/filter/main.go -addr "localhost:8081" -ingresshost "vald-ingress-filter.default.svc.cluster.local" -ingressport 8082 -egresshost "vald-egress-filter.default.svc.cluster.local" -egressport 8083
    ```
```

vankichi commented on example/client/gateway/filter/README.md:
````diff
@@ -0,0 +1,73 @@
+```
+
+5. Cleanup
+```bash
+helm uninstall vald
+kubectl delete -f ./example/manifest/filter/egress/deployment.yaml -f ./example/manifest/filter/egress/service.yaml -f ./example/manifest/filter/ingress/deployment.yaml -f ./example/manifest/filter/ingress/service.yaml
+```
````
```suggestion

    ```bash
    helm uninstall vald
    kubectl delete -f ./example/manifest/filter/egress/deployment.yaml -f ./example/manifest/filter/egress/service.yaml -f ./example/manifest/filter/ingress/deployment.yaml -f ./example/manifest/filter/ingress/service.yaml
    ```
```

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Update example/client/gateway/filter/README.md

Co-authored-by: Kiichiro YUKAWA <kyukawa315@gmail.com>

hlts2 committed:
Update example/client/gateway/filter/README.md

Co-authored-by: Kiichiro YUKAWA <kyukawa315@gmail.com>

hlts2 committed:
Update example/client/gateway/filter/README.md

Co-authored-by: Kiichiro YUKAWA <kyukawa315@gmail.com>

hlts2 committed:
Update example/client/gateway/filter/README.md

Co-authored-by: Kiichiro YUKAWA <kyukawa315@gmail.com>

hlts2 committed:
Update example/client/gateway/filter/README.md

Co-authored-by: Kiichiro YUKAWA <kyukawa315@gmail.com>

ykadowak commented on pkg/gateway/filter/handler/grpc/handler.go:
```diff
@@ -1434,47 +1434,57 @@ func (s *server) Search(ctx context.Context, req *payload.Search_Request) (res *
+			q := &payload.Filter_Query{
+				Query: filterConfig.Query,
+			}
+			d, err := c.FilterDistance(ctx, &payload.Filter_DistanceRequest{
```
Here `FilterDistance` is applied but is `FilterVector` applied to the search result anywhere?

ykadowak commented on example/server/egress-filter/main.go:
```diff
@@ -0,0 +1,122 @@
+}
+
+func (s *myEgressServer) FilterVector(ctx context.Context, in *payload.Filter_VectorRequest) (*payload.Filter_VectorResponse, error) {
+	// Write your own logic
+	glg.Logf("filtering the vector %#v", in)
+	return &payload.Filter_VectorResponse{
+		Vector: in.GetVector(),
+	}, nil
+}
```
How do you actually filter a vector? For example, if you want to filter out the `in.GetVector()`, how do you do it? Just pass an empty vector to the `Filter_VectorResponse`?

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: update schema and format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: schema and proto update and format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 commented:
@vankichi @ykadowak
Thank you for your previous review 🙏
Sorry, I updated the license header and the approval disappeared....
There is no change to the logic.

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Add Filter query configuration to Continuous Benchmark Job (#2296)

* fix: add filter query config to filter config

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

* fix: update benchmark job schema

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

* fix: make format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

---------

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

vankichi committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: makge changes to build ci-container

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Revert "fix: makge changes to build ci-container"

This reverts commit 24239503a56bfab301bea1f191b272664fca279d.

hlts2 committed:
fix: error of ci and add comment about import path

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

fix: format-diff ci error

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: make format and reviewdog warning

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: add new line for debug

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Revert "fix: add new line for debug"

This reverts commit ffecb7e7308f92aa67d99bda70e21cce7e5e299d.

hlts2 committed:
fix: use glg fatal function

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Update example/server/egress-filter/main.go

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Hiroto Funakoshi <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Update example/client/gateway/filter/main.go

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Hiroto Funakoshi <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: deleted unnecessary comment and variable

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: fix coderabbit warning

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: add query configuration to client-api-confi.md

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 commented on example/server/egress-filter/main.go:
```diff
@@ -0,0 +1,119 @@
+		glg.Fatal(err)
+	}
+
+	s := grpc.NewServer()
```
Since this is the example code, there is no need to set up a certificate.

hlts2 commented on example/server/egress-filter/main.go:
```diff
@@ -0,0 +1,119 @@
+	dimension        uint
+)
+
+func init() {
```
This is not a particular problem in this case, since it is just setting the flags for the command line application.

hlts2 commented on example/server/egress-filter/main.go:
```diff
@@ -0,0 +1,119 @@
+	egress.UnimplementedFilterServer
+}
+
+func (s *myEgressServer) FilterDistance(ctx context.Context, in *payload.Filter_DistanceRequest) (*payload.Filter_DistanceResponse, error) {
+	glg.Log("filtering vector %#v", in)
+	qCategory, ok := getSplitValue(in.GetQuery().GetQuery(), "=", 1)
+	if !ok {
+		return &payload.Filter_DistanceResponse{
+			Distance: in.GetDistance(),
+		}, nil
+	}
+
+	filteredDis := []*payload.Object_Distance{}
+	for _, d := range in.GetDistance() {
+		iCategory, ok := getSplitValue(d.GetId(), "_", 1)
+		if !ok {
+			continue
+		}
+		glg.Infof("qCategory: %v, iCategory: %v", qCategory, iCategory)
+		if qCategory == iCategory {
+			filteredDis = append(filteredDis, d)
+		}
+	}
+
+	if len(filteredDis) == 0 {
+		return nil, status.Error(codes.NotFound, "FilterDistance results not found.")
+	}
+
+	return &payload.Filter_DistanceResponse{
+		Distance: filteredDis,
+	}, nil
+}
```
If there is no category information in the query, the search results should be returned as is without returning an error.
Therefore, the existing implementation is fine.

hlts2 commented on docs/user-guides/client-api-config.md:
```diff
@@ -66,12 +66,13 @@
-			Filters: &payload.Filter_Config{
-				Targets: []*payload.Filter_Target{
-					{
+			Filters: []*payload.Filter_Config{
+				{
+					Target: &payload.Filter_Target{
 						Host: "vald-ingress-filter",
 						Port: 8081,
 					},
+					Query: &payload.Filter_Query{},
```
There is no need to configure Filter Query in this section. It is described elsewhere.

hlts2 commented on docs/user-guides/client-api-config.md:
```diff
@@ -16,13 +16,13 @@ It requires the vector, its ID (specific ID for the vector), and optional config
 message Config {
-  // Check whether or not the same set of vector and ID is already inserted.
+  // A flag to skip exist check during insert operation.
   bool skip_strict_exist_check = 1;
-  // Configuration for filters if your Vald cluster uses filters.
-  Filter.Config filters = 2;
-  // The timestamp when the vector was inserted.
+  // Filter configurations.
+  repeated Filter.Config filters = 2;
+  // Insert timestamp.
```
Yes, could you please create draft descriptions for this field?

hlts2 committed:
fix: add filter configuration to api docs

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

deepsource-autofix[bot] committed:
style: format code with Gofumpt and Prettier

This commit fixes the style issues introduced in 799a5f1 according to the output
from Gofumpt and Prettier.

Details: https://github.com/vdaas/vald/pull/2097

hlts2 committed:
fix: update filter-gateway api document

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

deepsource-autofix[bot] committed:
style: format code with Gofumpt and Prettier

This commit fixes the style issues introduced in 266f35b according to the output
from Gofumpt and Prettier.

Details: https://github.com/vdaas/vald/pull/2097

hlts2 commented on docs/api/insert.md:
```diff
@@ -53,11 +53,11 @@ Inset RPC is the method to add a new single vector.
+    | :---------------------: | :----------------------------- | :---- | :------: | :------------------------------------------------------------------------------------------------------------ |
+    | skip_strict_exist_check | bool                           |       |          | Check whether the same vector is already inserted or not.<br> The ID should be unique if the value is `true`. |
+    |        timestamp        | int64                          |       |          | The timestamp of the vector inserted.<br>If it is N/A, the current time will be used.                         |
+    |         filters         | repeated(Array[Filter.Config]) |       |          | Configuration for filter.                                                                                     |
```
It is better to be unified with the existing way of writing, so there is no need to modify it.

hlts2 commented on docs/api/upsert.md:
```diff
@@ -57,12 +57,12 @@ Upsert RPC is the method to update the inserted vector to a new single vector or
+    | :---------------------: | :----------------------------- | :---- | :------: | :------------------------------------------------------------------------------------------------------------ |
+    | skip_strict_exist_check | bool                           |       |          | Check whether the same vector is already inserted or not.<br> The ID should be unique if the value is `true`. |
+    |        timestamp        | int64                          |       |          | The timestamp of the vector inserted.<br>If it is N/A, the current time will be used.                         |
+    |         filters         | repeated(Array[Filter.Config]) |       |          | Configuration for filter.                                                                                     |
```
It is better to be unified with the existing way of writing, so there is no need to modify it.

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
fix: make proto/all format

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: build error

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
fix: resolve deprecated code

Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query

hlts2 committed:
Merge branch 'main' into feature/egress-gateway/add-egress-query
